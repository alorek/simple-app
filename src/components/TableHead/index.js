import React, { Component } from 'react';
import { titleize } from '../../utilities/textUtils';
import TableHeader from '../TableHeader';

class TableHead extends Component {

    renderColumns() {
        return this.props.dataKeys.map(key => {
            const column = {
                label: key === 'viewable' ? '' : titleize(key.replace('_', ' ')),
                id: key
            };
            return <TableHeader key={key} column = { column } />
        });
    }

    render() {

        return (

            <thead>
                <tr>
                    { this.renderColumns() }
                </tr>
            </thead>
        );
    }
}

export default TableHead;
