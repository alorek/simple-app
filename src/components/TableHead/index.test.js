import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import TableHead from './index';
import reducer from '../../reducers';

const store = createStore(reducer);

test('TableHead renders correctly', () => {

    const dataKeys = ['resource_type'];

    const tree = renderer.create(
        <Provider store={ store }>
            <TableHead dataKeys={ dataKeys } />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
