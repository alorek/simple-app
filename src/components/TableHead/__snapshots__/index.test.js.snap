// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead renders correctly 1`] = `
<thead>
  <tr>
    <th
      onClick={[Function]}
    >
      Resource Type
      <span
        className="glyphicon glyphicon-triangle-top"
      />
    </th>
  </tr>
</thead>
`;
